### DISCUSSION TITLE
Merging two `TypedDict` into one `dict[K, V]`

### DISCUSSION THREAD

**Discussion by moosbruggerj** [python/typing#1275] [Q&A] [answered]
Hi,
I currently have a piece of code, that receives two incompatible typed dicts, and I need to pass them as one merged dictionary along.
Preferably, I would like to use the `|` operator to merge the two dictionaries, however, this raises type errors in both mypy and pyright.

If I use unpacking, both type checkers accept the code.

```python
from typing import TypedDict
from typing_extensions import reveal_type

class A(TypedDict):
    x: str
    z: str

class B(TypedDict):
    x: str
    y: str

a: A = {'x': 'a', 'z': 'a'}
b: B = {'x': 'b', 'y': 'b'}
c = a | b  # <-- error: Operator "|" not supported for types "A" and "B"
reveal_type(c)  # pyright: unknown, mypy:  TypedDict('typed_dict_merge.A', {'x': builtins.str, 'z': builtins.str})
d = {**a, **b}
reveal_type(d)  # pyright: dict[str, str], mypy: builtins.dict[builtins.str, builtins.object]
print(c == d)  # prints True at runtime
```

Is there a way to annotate the code in a way to use the `|` operator? Preferably without using `cast`.
Explicitly annotating `c` does not seem to work:
```python
c: dict[str, str] = a | b  # error still persists
```

Creating a compatible `TypedDict` seems not to be a solution either. It only works with unpacking and pyright:
```python
class C(A, B, TypedDict):
    pass

c: C = a | b  # same as before
e: C = {'x': 'b', 'z': 'a', 'y': 'b'}  # ok
d: C = {**a, **b}  # ok in pyright, error in mypy: Expected TypedDict key to be string literal
reveal_type(d)  # revealed as C
```

**Context**
This piece of code comes from a django webserver and the two `TypedDicts` passed into this method contain data from different sources. The merged dictionary is passed along to django factory methods. This makes exact type information (`TypedDicts`) nice, but not necessary. However, the codebase uses the `|` operator to merge dicts a lot. Using unpacking here would stand out.

**Version**
python: `Python 3.10.8`
pyright: `pyright 1.1.276`
mypy: `mypy 0.961 (compiled: yes)`

**Comment by Gobot1234** [comment, marked as answer]
I think the ideal solution here is just to wait for Intersection to be finished (see #213)
